src/_data/glossary.yml

```diff
@@ -213,6 +213,47 @@
     - "language"
     - "type system"
 
+- term: "Code asset"
+  short_description: |-
+    Compiled native code that is bundled with a Dart app using a build hook
+    and can be used through `dart:ffi`.
+  long_description: |-
+    A **code asset** is compiled code that isn't written in Dart but
+    can be packaged and used by a Dart application.
+    Examples include shared libraries (`.so`, `.dll`, `.dylib`) built
+    in languages like C, C++, or Rust.
+
+    Code assets can be bundled with a Dart package using a [build hook][].
+    Dart code can call into these assets using [FFI][].
+
+    > Historically, code assets were also referred to
+    > as **native assets** or **native code assets**.
+
+    Code assets are useful for:
+
+    * Integrating with existing native libraries.
+    * Accessing system capabilities not directly exposed in Dart.
+    * Sharing performance-critical functionality across platforms.
+
+    For example, a Dart application could include a native `.so` library
+    that exposes specific functions, and call those functions from Dart.
+
+    [build hook]: /tools/hooks
+    [FFI]: /interop/c-interop
+  related_links:
+    - text: "C interop using dart:ffi"
+      link: "/interop/c-interop/"
+      type: "doc"
+    - text: "Hooks"
+      link: "/tools/hooks"
+      type: "doc"
+  labels:
+    - "language"
```

Comment:
On a second look through, I'm not sure the `language` label fits this one, as it's mostly a library and implementation feature.

Perhaps there's a new label we can put hook and asset related terms under? Maybe just "assets" or "build"? Do you have any thoughts?